### PR TITLE
[MONDRIAN-2630] Mondrian: filter a set of members including null, plu…

### PR DESCRIPTION
**Pull request by LeonardoCoelho71950**
…s a numeric filter, results in incorrect data

@pentaho/tatooine 
@pentaho-lmartins @ssamora 

@mkambol @lucboudreau

### CODE REVIEW

**Comment by LeonardoCoelho71950** [comment]
Thanks @mkambol. Working on the test.

File: mondrian/src/main/java/mondrian/rolap/SqlConstraintUtils.java [hunk 1]
```diff
@@ -166,67 +166,61 @@ public static void addContextConstraint(
...
-            {
-                sqlQuery.addWhere(expr, " is ", RolapUtil.sqlNullLiteral);
-            } else {
-                if (column.getDatatype().isNumeric()) {
-                    // make sure it can be parsed
-                    Double.valueOf(value);
-                }
+            if (column.getDatatype().isNumeric()) {
+                // make sure it can be parsed
+                Double.valueOf(value);
```
Comment (bennychow) [inline mondrian/src/main/java/mondrian/rolap/SqlConstraintUtils.java]: If a numeric column contained a null value, this line of code would throw a number format exception.  Would this cause any issues?  For example, in steelwheels, you can update CUSTOMER_W_TER.CREDITLIMIT to contain a null value.

Comment (LeonardoCoelho71950) [reply mondrian/src/main/java/mondrian/rolap/SqlConstraintUtils.java]: You're right, my integration test validates that scenario.

**Comment by LeonardoCoelho71950** [comment]
@mkambol, I added a test with the report we found yesterday.

**Comment by buildguy** [comment]
### :gear: Build finished in 3m 39s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -P !assemblies -pl mondrian
```
:exclamation: No tests found!


:information_source:  _This is an automatic message_

**Pull request merged**